webpack/components/Bookmark/AddBookmarkModal.js

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Modal, ModalVariant, TextInput,
+  Checkbox, Form, FormGroup,
+  ActionGroup, Button,
+} from '@patternfly/react-core';
+import { useDispatch } from 'react-redux';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { createBookmark, getBookmarks } from './BookmarkActions';
+
+const AddBookmarkModal = ({ selectedItem, onClose, controller }) => {
+  const dispatch = useDispatch();
+  const [name, setName] = useState(selectedItem);
+  const [query, setQuery] = useState(selectedItem);
+  const [isPublic, setIsPublic] = useState(false);
+  const submitDisabled = !name || !query;
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    dispatch(createBookmark({
+      controller, name, query, public: isPublic,
+    }, () =>
+      dispatch(getBookmarks(controller))));
+    onClose();
+  };
+
+  return (
+    <Modal
+      title={__('Add Bookmark')}
+      variant={ModalVariant.small}
+      isOpen
+      onClose={onClose}
+      appendTo={document.body}
+    >
+      <Form onSubmit={onSubmit}>
+        <FormGroup label={__('Name')} isRequired fieldId="name">
+          <TextInput
+            isRequired
+            type="text"
+            id="name"
+            aria-label="input_name"
+            name="name"
+            value={name}
+            onChange={setName}
+          />
+        </FormGroup>
+        <FormGroup label={__('Search Query')} isRequired fieldId="query">
+          <TextInput
+            isRequired
+            type="text"
+            id="query"
+            aria-label="input_query"
+            name="query"
+            value={query}
+            onChange={setQuery}
+          />
+        </FormGroup>
+        <FormGroup fieldId="latest">
+          <Checkbox
+            style={{ marginTop: 0 }}
+            id="public"
+            name="public"
+            label={__('Public')}
```

Comment:
What does "Public" mean?

Reply:
No idea! Isn't feature parody great!

Reply:
In all seriousness though.. I truthfully don't know...

Reply:
I like this addition, will do!